### ISSUE TITLE
`unnecessary-pass` (`PIE790`) only checks for docstrings immediately followed by a `pass` statement

### ISSUE THREAD

**Issue by tjkuson** [astral-sh/ruff#7600]
Currently, `PIE790` triggers on

```python
def foo():
    """A docstring."""
    pass
```

as expected. However, it does not trigger on

```python
def foo():
    print("foo")
    pass
```

or

```python
def foo():
    """A docstring."""
    print("foo")
    pass
```

which is unexpected.

**Comment by dhruvmanila** [comment]
I think that's an expected behavior as per the implementation. It's detected only when a docstring is followed by the pass statement. Do you think the [documentation](https://docs.astral.sh/ruff/rules/unnecessary-pass/) was unclear regarding this?

**Comment by tjkuson** [comment]
> Do you think the [documentation](https://docs.astral.sh/ruff/rules/unnecessary-pass/) was unclear regarding this?

Yes. Perhaps it's just me, but I think the documentation suggests that it catches all unnecessary `pass` statements, with a `pass` statement immediately following a docstring as _one example_ of an unnecessary `pass` statement. If this really is intended, then I think the documentation (and the name of the rule) should be altered.

However, might there be merit to changing the implementation to catch _all_ unnecessary `pass` statements? It would be more befitting of the current rule name and probably more useful to users.

**Comment by dhruvmanila** [comment]
> However, might there be merit to changing the implementation to catch _all_ unnecessary `pass` statements? It would be more befitting of the current rule name and probably more useful to users.

Yeah, I'm thinking in the same direction. I think the rule should be updated to catch all such cases.

It should be fairly simple to implement this as the function body is just a `Vec<Stmt>`, so we could just check the length of it. If the number of statements is greater than 1 and the last statement is a `Pass` statement then we can just remove it.

**Comment by charliermarsh** [comment]
Yeah, I don't mind changing it.
